sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java

```diff
@@ -380,6 +394,34 @@ public void run() {
         });
     }
 
+    private synchronized void instanceSignOut() {
+        if (mTokenStorage.getToken() == null) {
+            AppCenterLog.warn( LOG_TAG, "Couldn't sign-out: authToken doesn't exist.");
+            return;
+        }
+        mSignInDelayed = false;
+        removeAccount(mTokenStorage.getHomeAccountId());
+        mTokenStorage.removeToken();
+        AppCenterLog.info(LOG_TAG, "User sign-out succeeded.");
+    }
+
+    private void removeAccount(final String homeAccountIdentifier) {
+        if (mAuthenticationClient == null || homeAccountIdentifier == null) {
+            return;
+        }
+        mAuthenticationClient.getAccounts(new PublicClientApplication.AccountsLoadedListener() {
+
+            @Override
+            public void onAccountsLoaded(List<IAccount> accounts) {
+                for (IAccount account : accounts) {
+                    if (account.getHomeAccountIdentifier().getIdentifier().equals(homeAccountIdentifier)) {
+                        mAuthenticationClient.removeAccount(account);
```

Comment:
threading unsafe here, mAuthenticationClient should be accessed from worker thread (it mutates in worker thread) AND checking we are still enabled.

To do that simply wrap the entire code of the callback into
```
post(new Runnable() {

     @Override
     public void run() {
            for (IAcccont....

     }
}
```